chronos/streams/asyncstream.nim

```diff
@@ -24,15 +24,15 @@ const
     ## AsyncStreamWriter leaks tracker name
 
 type
-  AsyncStreamError* = object of CatchableError
+  AsyncStreamError* = object of TransportError
```

Comment:
I'm not sure it's a good idea to make this a transport error, ie this is a different abstraction level and the inheritance creates an abstraction leak.. 

better put the transport error in the `parent` field and translate, so that callers only ever have to understand `AsyncStreamError` and its derivatives for normal error handling

Reply:
Every `AsyncStream` needs source of data, there is no empty sourced AsyncStreams right now. So at the end every AsyncStream will pop up TransportError if transport fails to deliver data.

Reply:
sure - but the inheritance here seems unnecesary - ie it doesn't satisfy the usual "is a" test - async stream is not a transport, so why should its errors be transport errors?